### PR TITLE
new versions of make_requests and handle_responses that use apollo-federation's Connector

### PR DESCRIPTION
**Pull request by lennyburdette**
After we switch the router services to use Connectors generated in expand_connectors(), we can start using these functions and delete:

* src/plugins/connectors/response_formatting/ — will be replaced with apply_with_selection
* src/plugins/connectors/finder_field.rs — not necessary in single-phase query planning!
* src/plugins/connectors/http_json_transport.rs — saving the part that use the new Transport
* src/plugins/connectors/request_inputs.rs — didn't need to be its own module
* src/plugins/connectors/request_response.rs

This version leaves out some functionality:

* instrumentation, logging, and json selection diagnostics
* alias and __typename handling (beyond the root output type)

we should address both of these when we add JSONSelection::apply_with_selection or whatever we're going to call it

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Review by nicholascioli — APPROVED** [review]
Left a few nits, but it looks good!

File: apollo-router/src/plugins/connectors/handle_responses.rs [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+        let body = &hyper::body::to_bytes(body)
+            .await
+            .map_err(|_| InvalidResponseBody("couldn't retrieve http response body".into()))?;
+
+        if parts.status.is_success() {
+            let json_data: Value = serde_json::from_slice(body)
+                .map_err(|_| InvalidResponseBody("couldn't deserialize response body".into()))?;
+
+            let mut res_data = {
+                let (res, _apply_to_errors) = connector.selection.apply_to(&json_data);
```
[the remark this answers is not in the document]
Comment (lennyburdette) [reply apollo-router/src/plugins/connectors/handle_responses.rs:63]: for now, we'll revisit when we have the selection-set-aware version of apply_to. they're not really errors, they're diagnostics we can log to help customers understand when the JSON selection doesn't match the response

[the remark this answers is not in the document]
Comment (lennyburdette) [reply apollo-router/src/plugins/connectors/handle_responses.rs:63]: i can put a todo

File: apollo-router/src/plugins/connectors/handle_responses.rs [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+                    .extension("connector", connector.id.label.clone())
+                    .build(),
+            );
+        }
+    }
+
+    let data = match data.is_empty() {
+        true => None,
+        false => Some(Value::Object(data)),
+    };
```
[the remark this answers is not in the document]
Comment (lennyburdette) [reply apollo-router/src/plugins/connectors/handle_responses.rs:155]: dope, i knew there would be something better!

File: apollo-router/src/plugins/connectors/make_requests.rs [hunk 3]
```diff
@@ -0,0 +1,1280 @@
...
+                    .type_condition
+                    .as_ref()
+                    .ok_or_else(|| InvalidOperation("missing type condition".into()))?;
+                Ok(frag
+                    .selection_set
+                    .iter()
+                    .map(|sel| {
+                        let field = match sel {
+                            apollo_compiler::ast::Selection::Field(f) => f,
+                            apollo_compiler::ast::Selection::FragmentSpread(_) => todo!(),
```
Comment (lennyburdette) [inline apollo-router/src/plugins/connectors/make_requests.rs:337]: TODO!

File: apollo-federation/src/sources/connect/mod.rs [hunk 4]
```diff
@@ -60,3 +62,25 @@ impl Display for ConnectId {
         write!(f, "{}", self.label)
     }
 }
+
+impl ConnectId {
+    pub fn new_for_test(subgraph_name: NodeStr, type_name: Name, field_name: Name) -> Self {
```
[the remark this answers is not in the document]
Comment (lennyburdette) [reply apollo-federation/src/sources/connect/mod.rs:67]: does that work across crates? i'm getting `no function or associated item named `new_for_test` found for struct `ConnectId` in the current scope`

**Pull request merged**